### DISCUSSION TITLE
Build failed

### DISCUSSION THREAD

**Discussion by CMEONE** [openpgpjs/openpgpjs#1230] [Q&A]
```
[!] Error: Unexpected token (Note that you need @rollup/plugin-json to import JSON files)
../../../node_modules/elliptic/package.json (2:9)
1: {
2:   "_from": "elliptic@6.5.3",


npm ERR! code 1
npm ERR! path redacted
npm ERR! command failed
npm ERR! command sh -c rollup --config

npm ERR! A complete log of this run can be found in:
npm ERR!     redacted
npm ERR! code 1
npm ERR! path redacted
npm ERR! command failed
npm ERR! command sh -c npm run build

npm ERR! A complete log of this run can be found in:
npm ERR!     redacted
```

I have done `npm i -g @rollup/plugin-json --save-dev` which has not worked. Any support on building would be much appreciated!

**Comment by twiss** [comment]
Hi :wave: Could you please provide a bit more context? What did you do, what command did you run, etc?

**Comment by CMEONE** [comment]
Hello, thanks for helping out!

I first ran `npm install && npm test` as stated in the README instructions at the bottom. Then, I ran into an error because I didn't have rollup installed, so i did `npm i -g rollup` and ran into this error. As the error says, it wasn't able to parse the JSON file in `../../../node_modules/elliptic/package.json` and needed the `@rollup/plugin-json` dependency, so I installed that with `npm i -g @rollup/plugin-json --save-dev`.

**Comment by CMEONE** [comment]
Even after installing `@rollup/plugin-json`, I still received the parsing error telling me to install `@rollup/plugin-json`.

**Comment by CMEONE** [comment]
Here are the last few lines of my verbose log:
```
81 timing auditReport:init Completed in 322ms
82 timing reify:audit Completed in 723ms
83 timing reify Completed in 1288ms
84 timing command:install Completed in 5308ms
85 verbose stack Error: command failed
85 verbose stack     at ChildProcess.<anonymous> (/Users/redacted/.nvm/versions/node/v15.1.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/index.js:64:27)
85 verbose stack     at ChildProcess.emit (node:events:327:20)
85 verbose stack     at maybeClose (node:internal/child_process:1055:16)
85 verbose stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:288:5)
86 verbose pkgid openpgp@4.10.10
87 verbose cwd /redacted/openpgpjs
88 verbose Darwin 19.0.0
89 verbose argv "/Users/redacted/.nvm/versions/node/v15.1.0/bin/node" "/Users/redacted/.nvm/versions/node/v15.1.0/bin/npm" "install"
90 verbose node v15.1.0
91 verbose npm  v7.0.8
92 error code 1
93 error path /redacted/openpgpjs
94 error command failed
95 error command sh -c npm run build
96 verbose exit 1
```

**Comment by twiss** [comment]
You shouldn't have to install rollup manually, as it's already listed in the dev dependencies in our package.json. Could you try starting over, (uninstalling rollup, deleting the openpgp.js directory and cloning it again), and running just `npm install`, and post the output of that?

**Comment by twiss** [comment]
Also, just as a guess, maybe you had the `NODE_ENV` environment variable set to `production`? If so, npm won't install dev dependencies (which you need to build). See https://stackoverflow.com/a/34719468.

**Comment by CMEONE** [comment]
I just uninstalled rollup and recloned, tried `npm install --only=dev` and `npm install --dev`, both returning the following error:

```

> openpgp@4.10.10 build
> rollup --config

sh: rollup: command not found
```

`NODE_ENV` is undefined, so it's not set to production.

**Comment by twiss** [comment]
What's your Node.js and npm version?

**Comment by CMEONE** [comment]
```
> node -v
v15.1.0
> npm -v
7.0.8
```

**Comment by CMEONE** [comment]
I've tried everything. Deleted `node_modules`, recloned, reinstalled, checked 5 different ways that I'm in dev and not in production, but it still won't get the dev dependencies installed and crashes when trying to parse the elliptic package.json saying to install `@rollup/plugin-json`, which I do (`npm i -g @rollup/plugin-json`, `npm i --save @rollup/plugin-json`, `npm i --save-dev @rollup/plugin-json`, nothing works). 

Any advice on how to proceed? I really want to use OpenPGP.js

**Comment by twiss** [comment]
You could grab a build from here, and include that in your app instead: https://unpkg.com/openpgp / https://unpkg.com/browse/openpgp/dist/
